tools/IoTEdgeDevOps/DevOpsLib/BugQuery.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace DevOpsLib
+{
+    using System;
+
+    public class BugQuery
+    {
+        public BugQuery(string area, BugPriorityGrouping bugPriorityGrouping, bool inProgress)
+        {
+            ValidationUtil.ThrowIfNullOrWhiteSpace(area, nameof(area));
+            ValidationUtil.ThrowIfNull(bugPriorityGrouping, nameof(bugPriorityGrouping));
+            ValidationUtil.ThrowIfNull(inProgress, nameof(inProgress));
+
+            this.Area = area;
+            this.BugPriorityGrouping = bugPriorityGrouping;
+            this.InProgress = inProgress;
+        }
+
+        public string Area { get; }
+        public BugPriorityGrouping BugPriorityGrouping { get; }
+        public bool InProgress { get; }
+
+        public string Title
+        {
+            get
+            {
+                string titleBase = $"{this.Area}-{this.BugPriorityGrouping.Name}";
+
+                if (this.InProgress)
+                {
+                    return $"{titleBase}-Started";
+                }
+                else
+                {
+                    return $"{titleBase}-Not-Started";
+                }
+            }
+        }
+
+        public string GetWiqlFromConfiguration()
+        {
+            string query;
+            string unassignedPriorityPlaceholder = "Other";
+
+            if (this.BugPriorityGrouping.Priority != unassignedPriorityPlaceholder && !this.InProgress)
```

Comment:
you can define a data collection type to store this mapping.
1. uses Dictionary<string, Dictionary<string, string>>, or
2. uses Dictionary<Tuple<string,string>, string>

Then use TryGetValue if value doesn't exist, throw exception.

Reply:
I don't think this is a good idea because then we have to enumerate all combinations, where if/else chain can capture multiple combinations with the same logic.

Reply:
you don't need to enumerate all combinations.  For example Dictionary<tuple<string,bool>, string>

if-eles chain block can be replaced by:

if (!this.Dict.TryGetValue(Tuple.Create(this.BugPriorityGrouping.Priority, this.InProgress), out string query) {
    throw new NotImplementedException();
}

query = query.Replace("{PRIORITY}", this.BugPriorityGrouping.Priority.ToString());
query = query.Replace("{AREA}", this.Area);

is it clear to you?

Reply:
This part made sense to me, but when making the dict we will have to enumerate all the mappings, right? Might be missing something obvious, so apologies if I am 😅.

Reply:
Ok.  Got it because current if-else block is not just use equals, but not equals; therefore need to define more than 4 entry in the dictionary.  my fault I missed that.

Why not nested the if statement? for example you can nested 1st and 2nd if block, right?